torch/nn/parallel/replicate.py

```diff
@@ -80,6 +80,9 @@ def replicate(network, devices, detach=False):
         raise RuntimeError("Cannot replicate network where python modules are "
                            "childrens of ScriptModule")
 
+    if not devices:
```

Comment:
do we still need this? Looks like there is no valid way to trigger this as `self.device_ids[:len(inputs)]` will at least contain one device?

Reply:
Prefer to keep this. It is a sanity check to prevent potential crash when devices is empty, though it won't be trigger for now.